index.php

```diff
@@ -9,4 +9,135 @@
+    wp_register_script(
+        'feedback-fish',
+        "https://feedback.fish/ff.js?pid=$projectId",
+        [],
+        null,
+        true
+    );
+    wp_enqueue_script('feedback-fish');
+}
+add_action('wp_enqueue_scripts', 'enqueue_feedback_fish_script');
+
+// Add defer="defer" to the feedback.fish <script> tag to avoid blocking render
+// Taken from: https://wordpress.stackexchange.com/a/38335
+add_filter(
+    'script_loader_tag',
+    function ($tag, $handle) {
+        if ('feedback-fish' !== $handle) {
+            return $tag;
+        }
+
+        return str_replace(' src', ' defer="defer" src', $tag);
+    },
+    10,
+    2
+);
+
+// Add "Send feedback" to the primary nav if it exists
+function add_feedback_fish_nav_menu_item($items, $args)
+{
+    $current_user = wp_get_current_user();
+    $manual_usage = get_option('feedback_fish_manual');
+    if (!$manual_usage) {
+        $items .= "<li class=\"menu-item\"><a href=\"#\" data-feedback-fish-userid=\"$current_user->user_email\" data-feedback-fish>Send feedback</a></li>";
+    }
+    return $items;
+}
+add_filter('wp_nav_menu_items', 'add_feedback_fish_nav_menu_item', 10, 2);
+
+// Add "Feedback Fish" settings page to the WPAdmin menu
+function create_feedback_fish_settings_page()
+{
+    $page_title = 'Feedback Fish Settings';
+    $menu_title = 'Feedback Fish';
+    $capability = 'manage_options';
+    $slug = 'feedback_fish';
+    $callback = 'feedback_fish_settings_page_content';
+
+    add_submenu_page(
+        'plugins.php',
+        $page_title,
+        $menu_title,
+        $capability,
+        $slug,
+        $callback
+    );
+}
+add_action('admin_menu', 'create_feedback_fish_settings_page');
+
+// General content for the settings page
+function feedback_fish_settings_page_content()
+{
+    ?>
+  <div class="wrap">
+    <h2>Feedback Fish</h2>
+    <form method="post" action="options.php">
+      <?php
+      settings_fields('feedback_fish');
+      do_settings_sections('feedback_fish');
+      submit_button();?>
+    </form>
+  </div>
+<?php
+}
+
+// Setup the settings page sections
+add_action('admin_init', 'setup_feedback_fish_settings_page_sections');
+add_action('admin_init', 'setup_feeback_fish_settings_fields');
+function setup_feedback_fish_settings_page_sections()
+{
+    add_settings_section(
+        'feedback_fish_general',
+        'General',
+        'feedback_fish_settings_page_section_callback',
+        'feedback_fish'
+    );
+}
+
+function setup_feeback_fish_settings_fields()
+{
+    add_settings_field(
+        'feedback_fish_project_id',
+        'Project ID',
+        'feedback_fish_project_id_field',
+        'feedback_fish',
+        'feedback_fish_general'
+    );
+    add_settings_field(
+        'feedback_fish_manual',
+        'Manual Usage (advanced)',
+        'feedback_fish_manual_field',
+        'feedback_fish',
+        'feedback_fish_general'
+    );
+    register_setting('feedback_fish', 'feedback_fish_project_id');
```

Comment:
I imagine there's a way to validate that the ID is for an existing Feedback Fish project? I'd recommend passing a `sanitize_callback` function where you can validate that the user added a valid project id, and if they did not you could return a validation message. 

Like if I enter: `fake project` as the project ID, you probably don't want to enqueue that to the users front end if you could instead tell the user when they save the setting that `hey pal, this project ID is invalid`

So, in your sanitize_callback you could, if it's possible, make a remote HTTP request to your service to validate if the project ID exists or not and if not, tell the user on the settings page before the option is saved.

Reply:
In addition to validation, you also want the `sanitize_callback` to ensure users are putting safe content in the database. 

See: https://codex.wordpress.org/Validating_Sanitizing_and_Escaping_User_Data#Sanitizing:_Cleaning_User_Input